code/game/machinery/machinery.dm

```diff
@@ -406,6 +406,8 @@ Class Procs:
 	if(ghost_read)
 		ghost_flags |= PERMIT_ALL
 	if(canGhostRead(usr,src,ghost_flags))
+		if(!isAdminGhost(usr)) //probably the laziest fix late night coding has ever seen
```

Comment:
actually this is firing even for ghosts opening APCs and stuff so I'll fix this tomorrow